Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -260,6 +262,34 @@ static ExceptionOr<void> checkDuplicateViewTransitionName(const AtomString& name
     return { };
 }
 
+static RefPtr<ImageBuffer> snapshotNodeVisualOverflowClippedToViewport(LocalFrame& frame, Node& node, SnapshotOptions&& options, LayoutRect& oldOverflowRect)
```

Comment:
I don't have a strong opinion but this could also go into FrameSnapshotting.cpp, also fine in ViewTransition.cpp to me. Up to you.